src/common/UserAvatar.js

```diff
@@ -20,32 +24,36 @@ type Props = $ReadOnly<{|
  * @prop [children] - If provided, will render inside the component body.
  * @prop [onPress] - Event fired on pressing the component.
  */
-export default class UserAvatar extends PureComponent<Props> {
-  render() {
-    const { avatarUrl, children, size, onPress } = this.props;
-    const borderRadius = size / 8;
-    const style = {
-      height: size,
-      width: size,
-      borderRadius,
-    };
+const UserAvatar = function UserAvatar(props: Props) {
+  const { avatarUrl, children, size, onPress } = props;
+  const borderRadius = size / 8;
+  const style = {
+    height: size,
+    width: size,
+    borderRadius,
+  };
 
-    return (
-      <View>
-        <Touchable onPress={onPress} style={style}>
-          <ImageBackground
-            style={style}
-            source={{
-              uri: avatarUrl.get(PixelRatio.getPixelSizeForLayoutSize(size)).toString(),
-            }}
-            resizeMode="cover"
-            /* ImageBackground seems to ignore `style.borderRadius`. */
-            borderRadius={borderRadius}
-          >
-            {children}
-          </ImageBackground>
-        </Touchable>
-      </View>
-    );
-  }
-}
+  const auth = useSelector((state: GlobalState) => getAuth(state));
+
+  return (
+    <View>
+      <Touchable onPress={onPress} style={style}>
+        <ImageBackground
+          style={style}
+          source={{
+            uri: avatarUrl.get(PixelRatio.getPixelSizeForLayoutSize(size)).toString(),
+            // For `FallbackAvatarURL`s.
+            headers: getAuthHeaders(auth),
```

Comment:
I'm puzzled now at why this hasn't applied all along to uploaded avatars. Discussion [in chat](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/302.20redirect.20for.20avatar/near/1090298).